nodes/modifier_make/cut_object_by_surface.py

```diff
@@ -117,51 +125,82 @@ def remember_connection(edges_by_face, bm_edge, bm_intersection):
 
             # We are using bvh's raycast to calculate intersections of object's edges
             # with the surface.
-            bvh = mathutils.bvhtree.BVHTree.FromPolygons(surf_vertices, surf_faces)
+            bvh = mathutils.bvhtree.BVHTree.FromPolygons(surf_vertices, surf_faces, all_triangles = self.triangles)
             bm_obj = bmesh_from_pydata(obj_vertices, obj_edges, obj_faces)
             bm_cut = bmesh.new()
             edges_by_face = defaultdict(list)
             bm_obj.verts.index_update()
             bm_obj.edges.index_update()
             for bm_edge in bm_obj.edges:
                 v1, v2 = bm_edge.verts
-                edge = v1.co - v2.co
+                edge = v2.co - v1.co
```

Comment:
can you explain why `v1.co - v2.co` was incorrect?  are edge indices sorted by distance to zero vector?

Reply:
It was not incorrect, it just that if you do `v1.co - v2.co`, then when raycasting from `v1.co` the direction to project is `-edge`, and it seems more logical and neat to me to indeed project from `v1.co`, but along `edge` direction, so I did `v2.co - v1.co`.